### ISSUE TITLE
UI Improvements ( Album View / Background Player )

### ISSUE THREAD

**Issue by annomatik** [kaaholst/android-squeezer#724]
This is a crosspost of https://github.com/nikclayton/android-squeezer/issues/686 , as I don't know, what's the current github for that project.

Hi! I have two feature requests for UI improvements. In BlackPlayer, there is a very nice album view which uses the display space of my mobile phone nicely and looks good, I've attached a picture. Would it be possible to have something similar in Squeezer two?

Also, BlackPlayer has "background player controls", for the lack of a better description. That is, regardless, in which app I am, I can just drop down from the top of my phone the player controls, skip to the next / previous track there, play/pause the current track, see, how many tracks there are in my current playlist, together with track name, artist name and album art.

Both of these features would be a nice improvement to Squeezer. Thank you!

![](https://user-images.githubusercontent.com/28311452/128571163-a838420b-0ee3-4053-a09d-f10e231f5a76.jpg)

**Comment by kaaholst** [comment]
Hi and thanks for the feedback.

Regarding the Album view, did you try the grid layout (Large Artwork) in Squeezer, which is available from the 3-dot options menu in the right / View.

Currently in Squeezer you can drag down notifications where you can skip to next/previous, play/pause and see the currently playing song. Volume can be controlled via the hardware volume keys. This works whether Squeezer is in he foreground or another app is in foreground and even if the phone is locked. If you don't see the notification, you might have to change the notification settings for the "Squeezer Ongoing Notification".

The Squeezer notification doesn't show the current track number or the number of tracks in the current playlist. I can add that if you like.

![billede](https://user-images.githubusercontent.com/428304/128764903-1765be80-6a65-40de-abdd-326f5322934b.png)

![billede](https://user-images.githubusercontent.com/428304/128765867-c92cfcf0-4380-48ed-b285-c4f84ff8e49a.png)

![billede](https://user-images.githubusercontent.com/428304/128766358-f1e74070-4aa0-4e73-aba9-75381ef19466.png)

**Comment by annomatik** [comment]
Well, the large artwork option is a bit... large :-D

![Screenshot_20210809-221649_Squeezer](https://user-images.githubusercontent.com/28311452/128768860-eb1bc88e-9dcf-4aa2-ac86-bd9d80afc14b.jpg)

I can't find "Squeezer Ongoing Notification" -- is 2.2.7 the right version (installed from the Play Store)? Or do I have to update (from where?)?

**Comment by kaaholst** [comment]
2.2.7 is the right version. The notification shall always be shown when Squeezer is connected to a LMS, it is a requirement from Android. The notification can however be disabled via the phone settings.

I see what you mean about the size of the artwork. It is tricky to fix however, because I've had complains that were not showing enough of the title and/or artist if the size is reduced. I will see if I can work something out for your screen size.

Could you provide me the dimensions and resolution of the display on your phone?

**Comment by annomatik** [comment]
Sure, it's a [Samsung S7](https://www.gsmarena.com/samsung_galaxy_s7-7821.php). (1440x2560px)

**Comment by annomatik** [comment]
I looked into the Settings and yes, the Notifications were indeed turned off for Squeezer. Thanks, works nicely now!

It would be cool to have the same functions like in Blackplayer, that is a "x/n" Indication, track x of n from playlist and maybe even a "previous track" button, left of the play/pause. The lag of play/pause is probably a problem of LMS, right?

**Comment by annomatik** [comment]
By the way, maybe a new topic, but still... On my lock screen, I see three controls (Prev / Play/Pause / Next), but when I press them, nothing happens.

![Screenshot_20210810-194315_Squeezer](https://user-images.githubusercontent.com/28311452/128909097-c09e7465-c3cc-47ac-abce-5966c982836a.jpg)

**Comment by kaaholst** [comment]
Hi,

I reduced the size of the grid items for devices with smaller screens. There is no point in having a grid layout with one column!
I also added  the current track number or the number of tracks in the current playlist to the notification.

I updated the [apk](https://drive.google.com/file/d/1xanOkuQpy36v2aw4OIm1brkCriPII6s_/view?usp=sharing) from #723. That is, the link is the same, but the apk it points to is updated. You can try the new features from there.

The lag of play/pause depends on the server you run LMS on, what you are playing, your network, and the device which runs Squeezer, so yes it may differ.

On the notification there should be a small arrow on the top right indicating it can be expanded. If you press the arrow or drag down, it will show two additional buttons: disconnect and skip to previous.

On my phone (as well as my test devices / emulators) I can use the notification buttons on the lock screen. Maybe settings again, or maybe this is a Samsung "feature"?

From the Android documentation:
_Users can use the system settings to choose the level of detail visible in lock screen notifications, including the option to disable all lock screen notifications. Starting with Android 8.0, users can choose to disable or enable lock screen notifications for each notification channel._

**Comment by annomatik** [comment]
Hi! It's better now, there's 4 instead of just 2 albums on a screen. Would it be possible to make it even smaller? Blackplayer can fit 3 albums in a row which imho looks very good and works well, maybe you can also make the font-size a bit smaller?

![Screenshot_20210815-152231_Squeezer](https://user-images.githubusercontent.com/28311452/129480082-93eb79c2-4a17-4865-a72d-b49d03d3a65c.jpg)

![Screenshot_20210815-152206_BlackPlayer EX](https://user-images.githubusercontent.com/28311452/129480080-38ee25e5-0bb1-45f1-aab5-26881c38b134.jpg)

**Comment by kaaholst** [comment]
I agree that it looks good. I don't think it's good for usability though because to much of the title and album/artist may be cut off. Of course that may depend on the quality of the artwork, and the font size. 

For a screen of your size maybe the best is actually the list layout. 

I leave this open, for anyone to pick up.
